interop/reusability.md

```diff
@@ -0,0 +1,1088 @@
+This document illustrates two major goals for the property decorator
+proposal:
+
+1. It should be possible to write general-purpose decorators that work
+   across all of the different syntactic forms for creating declarative
+   properties in JavaScript (both classes and object literals).
+2. Decorators implement syntactic abstractions, and therefore receive
+   information about the syntactic form they are decorating.
+
+## Motivating Example
+
+To illustrate, I'll write a decorator called `reader` that decorates a
+property beginning with an `_` and creates a public getter for that
+property without an `_`.
+
+Here's the decorator definition:
+
+```js
+function reader(target, descriptor) {
+  let { enumerable, configurable, property: { name, get }, hint } = descriptor;
+
+  // extractPublicName('_first') === 'first'
+  let publicName = extractPublicName(name() /* extract computed property */);
+
+  // define a public accessor: get first() { return this._first; }
+  Object.defineProperty(target, publicName, {
+    // give the public reader the same enumerability and configurability
+    // as the property it's decorating
+    enumerable, configurable, get: function() { return get(this, name); }
+  });
+  
+  return descriptor;
+}
+
+function extractPublicName(name) {
+  // _first -> first
+  return name.slice(1);
+}
+```
+
+> For those of you thinking this is weak tea without true private state,
+> I have a treat for you in this repository: how this decorator could be
+> extended to work with hypothetical true private state. In the
+> meantime, bear with me.
+
+Next, we'll explore how this definition works on each kind of
+declarative property creation in JavaScript.
+
+## Basic Rules of Decorators
+
+Decorators always operate on a particular syntactic element,
+providing a hook into the runtime semantics for that syntax.
+
+If the runtime semantics for the syntax include "let x be the
+result of evaluating *SomeExpression*`, that expression is passed
```

Comment:
I believe this is mismatched tick/double quote. Probably was suppose to be either "let x be the result of evaluating *SomeExpression*" or `let x be the result of evaluating *SomeExpression*`